copilot-chat-git.el

```diff
@@ -102,111 +103,130 @@ repository or if there are no staged changes.
+                ;; Restore the git commit template comments and insert generated message
+                (goto-char (point-max))
+                (delete-region (point-min) (point-max))
+                (insert accumulated-content "\n\n" template-comments))
+              (message "Commit message generation completed.")
+              (copilot-chat--debug 'commit "Generation completed successfully")
+
+              ;; Remove the temporary instance from instances list to avoid memory leaks
+              (setq copilot-chat--instances (delq instance copilot-chat--instances)))
+
+          (progn
+            (when (string= accumulated-content "")
+              (goto-char start-pos)
+              (when (looking-at wait-prompt)
+                (delete-region start-pos (+ start-pos (length wait-prompt)))))
+
+            (setq accumulated-content (concat accumulated-content content))
+
+            (goto-char start-pos)
+            (delete-region start-pos (min (+ start-pos (length accumulated-content)) (point-max)))
+            (insert accumulated-content)
+            (copilot-chat--debug 'commit "Received chunk: %d chars" (length content))))))))
+
 ;;;###autoload (autoload 'copilot-chat-insert-commit-message-when-ready "copilot-chat" nil t)
 (defun copilot-chat-insert-commit-message-when-ready ()
-  "Generate and insert a commit message using Copilot.
-Uses the current staged changes in git
-to generate an appropriate commit message.
-Requires the repository to have staged changes."
+  "Generate and insert a commit message using GitHub Copilot.
+Uses the current staged changes in git to generate an appropriate commit message.
+Requires the repository to have staged changes ready for commit."
   (interactive)
-  ;; Get magit staged diff
   (aio-with-async
-    (let* ((instance (copilot-chat--current-instance))
+    (let* ((instance (copilot-chat--create-commit-instance))
            (current-buf (current-buffer))
            (start-pos (point))
-           (accumulated-content "")
-           (error-occurred nil)
            (diff (aio-await (copilot-chat--get-diff)))
-           (prompt (concat copilot-chat-commit-prompt diff))
-           ;; Store the git commit template comments
-           (template-comments
-            (save-excursion
-              (goto-char (point-min))
-              (let ((comments ""))
-                (while (re-search-forward "^#.*$" nil t)
-                  (setq comments (concat comments (match-string 0) "\n")))
-                comments)))
-           (wait-prompt "# [copilot-chat] Working on generating commit message, please wait... "))
-
-      ;; Debug messages using structured format
-      (copilot-chat--debug 'commit "Starting commit message generation")
+           (template-comments (copilot-chat--get-git-commit-template-comments))
+           (wait-prompt (format "# [copilot:%s] Generating commit message..." (copilot-chat-model instance)))
```

Comment:
This probably changes the loading indicator behavior, doesn't it?

It's not good practice to alter functionality in a PR labeled as a "refactor." People will review it assuming the behavior remains unchanged.

Looking back at the title, I see the type was actually "feat" (feature).

The better approach would be to first submit a PR that refactors the code without changing behavior, then follow up with a separate PR for the functionality changes.

Reply:
yes its fits better as a feat